src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  // static alive = [];
+
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  // static deleteDeadAnimal() {
+  //   Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  // }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(health, name);
+    this.hidden = false;
+  }
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    } else {
+      this.hidden = false;
+    }
```

Comment:
sounds logical to have the possibility set `hidden = false`, but it will not match with the method name (hide), so let's keep only the part where we set `hidden = true`